src/DOT/Dot.jl

```diff
@@ -4,17 +4,57 @@ using GraphIO.ParserCombinator.Parsers
 using LightGraphs
 using LightGraphs: AbstractGraphFormat
 
-import LightGraphs: loadgraph, loadgraphs
+import LightGraphs: loadgraph, loadgraphs, savegraph
 
 export DOTFormat
 
 struct DOTFormat <: AbstractGraphFormat end
-# TODO: implement save
+
+function savedot(io::IO, g::LightGraphs.AbstractGraph, gname::String = "")
+    isdir = LightGraphs.is_directed(g)
+    head = (isdir ? "digraph " : "graph ") * gname * " {"
+    nodes = ""
+    for i in LightGraphs.vertices(g)
+        nodes = nodes * "\n\t" * string(i) * " [label = " * string(i) * "]"
+    end
+    edg = ""
+    if isdir
+        for u in LightGraphs.vertices(g)
+            n = LightGraphs.outneighbors(g, u)
+            if length(n) == 0
+                continue
+            end
+            s = string(n)
+            edg = edg * "\n\t" * string(u) * " -> {" * s[2:length(s)-1] * "}"
+        end
+    else
+        for e in LightGraphs.edges(g)
+            source = string(LightGraphs.src(e))
+            dest = string(LightGraphs.dst(e))
+            edg = edg * "\n\t" * source * " -- " * dest
```

Comment:
The problem here is that you use `*` to build a large string. This creates a lot of unnecessary allocations and will become really slow when the graph gets big. Its better to work with IOBuffers or write each line directly to `io` as soon as you have finished creating it. In fact you can use `println(io, "some string")`, so you will not have to insert `\n` manually.